taqueria-vscode-extension/src/lib/gui/SandboxesDataProvider.ts

```diff
@@ -13,39 +14,84 @@ export class SandboxesDataProvider implements vscode.TreeDataProvider<SandboxTre
 	}
 
 	async getChildren(element?: SandboxTreeItem): Promise<SandboxTreeItem[]> {
-		let pathToDir: Util.PathToDir | null;
-		let config: Util.TaqifiedDir | null;
+		if (element) {
+			return [];
+		}
+		const [{ config }, runningContainers] = await Promise.all([this.getConfig(), getRunningContainerNames()]);
+		if (!config?.config?.sandbox) {
+			return [];
+		}
+		const environmentNames = this.getEnvironmentNames(config);
+
+		const getState = (sandBoxName: string) => {
+			if (runningContainers === undefined) {
+				return undefined;
+			}
+			for (const environmentName of environmentNames) {
+				if (runningContainers.findIndex(x => x === `taqueria-${environmentName}-${sandBoxName}`) >= 0) {
+					return true;
+				}
+			}
+			return false;
+		};
+		const sandboxNames = this.getSandboxNames(config);
+
+		return sandboxNames.map(sandboxName =>
+			new SandboxTreeItem(sandboxName, getState(sandboxName), vscode.TreeItemCollapsibleState.None)
+		);
+	}
+
+	private _onDidChangeTreeData: vscode.EventEmitter<SandboxTreeItem | undefined | null | void> = new vscode
+		.EventEmitter<
+		SandboxTreeItem | undefined | null | void
+	>();
+	readonly onDidChangeTreeData: vscode.Event<SandboxTreeItem | undefined | null | void> =
+		this._onDidChangeTreeData.event;
+
+	private async getConfig(): Promise<{ config: Util.TaqifiedDir | null; pathToDir: Util.PathToDir | null }> {
 		if (!this.workspaceRoot) {
-			pathToDir = null;
-			config = null;
+			return {
+				pathToDir: null,
+				config: null,
+			};
 		} else {
 			try {
-				pathToDir = await Util.makeDir(this.workspaceRoot, this.helper.i18n);
-				config = await Util.TaqifiedDir.create(pathToDir, this.helper.i18n);
+				const pathToDir = await Util.makeDir(this.workspaceRoot, this.helper.i18n);
+				const config = await Util.TaqifiedDir.create(pathToDir, this.helper.i18n);
+				return {
+					config,
+					pathToDir,
+				};
 			} catch (e: any) {
-				config = null;
 				await this.helper.notifyAndLogError('Error while loading config, sandboxes list will fail to populate', e);
+				return {
+					pathToDir: null,
+					config: null,
+				};
 			}
 		}
+	}
 
-		if (element) {
+	private getEnvironmentNames(config: Util.TaqifiedDir): string[] {
+		if (!config?.config?.environment) {
 			return [];
 		}
+		let environments = Object.entries(config.config.environment);
+		const defaultEnv = config.config.environment['default'];
+		environments = environments.filter(environment => environment[0] !== 'default');
```

Comment:
I feel like it's more readable to do the following:

`const environments = Object.entries(config.config.environment);` (from `let` to `const`)
Then
`const nonDefaultEnv = environments.filter(environment => environment[0] !== 'default');` (from `environments` to `const nonDefaultEnv`)

What do you think?